polar_route/crossing_smoothing.py

```diff
@@ -0,0 +1,1437 @@
+                cell['case'] = cell['case'][accessible_edges]
+                cell['neighbourIndex'] = cell['neighbourIndex'][accessible_edges]
+                cell['neighbourCrossingPoints'] = cell['neighbourCrossingPoints'][accessible_edges]
+                cell['neighbourTravelLegs'] = cell['neighbourTravelLegs'][accessible_edges]
+
+                self.dijkstra_graph[key] = cell
+
+    def _initialise_config(self):
+        '''    
+            Initialising configuration information. If None return a list of standards
+        '''
+
+        self.merge_separation = 1e-3
+        self.converged_sep    = 5e-3
+        self._g = pyproj.Geod(ellps='WGS84')
+
+    def _long_case(self,start,end,case,Sp,Cp,Np):
+        '''
+            Longitude based smoothing updating the crossing point given the conditions 
+            of the adjacency pair
+
+            Input:
+                start (dict) - Dictionary of the start cell information
+                end (dict)   - Dictionary of the end  cell information
+                case (int)   - Adjacency case type connecting the two cells
+                Sp (tuple)   - Start Point (long,lat)
+                Cp (tuple)   - Crossing Point (long,lat)
+                Np (tuple)   - End Point (long,lat)
+ 
+        '''
+        def NewtonOptimisationLong(f,y0,x,a,Y,u1,v1,u2,v2,speed_s,speed_e,R,λ_s,φ_r):
+                tryNum=1
+                iter=0
+                improving=True
+                _epsilon = 1e-4
+                while improving:  
+                    F,dF,X1,X2  = f(y0,x,a,Y,u1,v1,u2,v2,speed_s,speed_e,R,λ_s,φ_r)
+                    if (F==0) or (dF==0):
+                        dY = 0
+                    else:
+                        dY = (F/dF)
+                    if iter != 0:
+                        improving =  (abs(dY)>_epsilon) or (abs(dY) > _epsilon*(X1*X2) and (abs(dY)/iter) > _epsilon)
+                    else:
+                        improving = True
+                    y0  -= dY
+                    iter+=1
+
+                    if (iter>100 and tryNum == 1):
+                        y0 = Y*x/(x+a)
+                        tryNum+=1
+                    if (iter > 200) and tryNum>= 2 and tryNum < 10:
+                        tryNum+=1
+                        iter-=100
+                        if(Y < 0):
+                            if v2>v1:
+                                y0 = (tryNum-2)*Y
+                            else:
+                                y0 = (tryNum-3)*-Y
+                        else:
+                            if (v2<v1):
+                                y0 = (tryNum-2)*Y
+                            else:
+                                y0 = (tryNum-3)*-Y
+                    if iter > 1000:
+                        raise Exception('Newton Curve Issue - Longitude Case')
+                return y0
+
+        def _F(y,x,a,Y,u1,v1,u2,v2,speed_s,speed_e,R,λ_s,φ_r):
+            ρ = (λ_s+φ_r)/2.0
```

Comment:
I want to avoid using symbols in the codebase if possible. Can we give the parameters descriptive names in the codebase and explain which symbols the parameters map to in the paper in a docstring?

Reply:
Currently it would be good to keep symbols inside the codebase so that the maths lines up with the publications. Greek symbols should be pep-8 compliant.

Reply:
PEP8 does say `"Use non-ASCII characters sparingly, preferably only to denote places and human names."` but I'm not personally too bothered about leaving these in if it serves a purpose in supporting a publication and the parameters are properly defined in the docstring.